### PR TITLE
Refactor: underscores for private class methods and variables

### PR DESCRIPTION
**Pull request by JanGut**
This PR adds a lot of _underscores for private class methods and variables.
@Flova @NFiedler Please check the marked TODOs.

### CODE REVIEW

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py [hunk 1]
```diff
@@ -60,17 +60,17 @@ def __init__(self, config, model_path):
...
 
-        self.config = config
+        self._config = config
 
         # Setup detector
-        self.net = Detector(bytes(configpath, encoding="utf-8"), bytes(weightpath, encoding="utf-8"), 0.5, bytes(datapath, encoding="utf-8"))
+        self._net = Detector(bytes(configpath, encoding="utf-8"), bytes(weightpath, encoding="utf-8"), 0.5, bytes(datapath, encoding="utf-8"))
         # Set classes
-        self.classes = ["ball", "goalpost"]
+        self._classes = ["ball", "goalpost"]  # TODO Jan: this has no usages?! Why? Remove?
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]: See here...

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py [hunk 2]
```diff
@@ -141,29 +141,29 @@ def __init__(self, config, model_path):
         weightpath = os.path.join(model_path, "yolo_weights.weights")
         configpath = os.path.join(model_path, "config.cfg")
         # Set config
-        self.config = config
+        self._config = config
         # Define classes
-        self.classes = ["ball", "goalpost"]
+        self._classes = ["ball", "goalpost"]  # TODO Jan: this has no usages?! Why? Remove?
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]: And here...

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py [hunk 3]
```diff
@@ -261,55 +261,55 @@ def __init__(self, config, yolo):
...
         """
-        self.yolo.predict()
+        self._yolo.predict()
 
 class YoloGoalpostDetector(CandidateFinder):
     """
     A goalpost detector using the yolo neural network
     """
-    def __init__(self, yolo):
+    def __init__(self, yolo):  # TODO Jan: Why does this class have no config param? See YoloBallDetector...
```
Comment (JanGut) [inline bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]: As well as here...

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py (hunk 1 above)
Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]: Indeed this list is a usless development relict. The handler itself uses only the class_id.

File: bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py (hunk 3 above)
Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py]: The config param is currently unsued, but I added it for compatibility reasons.

File: bitbots_vision/src/bitbots_vision/vision_modules/color.py [hunk 4]
```diff
@@ -367,12 +364,12 @@ def __init__(self, config, package_path):
         super(DynamicPixelListColorDetector, self).__init__(config, package_path)
 
         # Annotate global variable. The global is needed due to threading issues
-        global dyn_color_space
-        dyn_color_space = np.copy(self.color_space)
+        global _dyn_color_space
```
Comment (Flova) [inline bitbots_vision/src/bitbots_vision/vision_modules/color.py]: Should we mark a `global` defined variable as private? I get the idea behind it, but it still seems very confusing.

Comment (JanGut) [reply bitbots_vision/src/bitbots_vision/vision_modules/color.py]: I looked it up in the google python code guidelines and understood this...

Comment (Flova) [reply bitbots_vision/src/bitbots_vision/vision_modules/color.py]: Okay I looked it up too and it looks okay.

**Pull request merged**